### PR TITLE
Added spoiler markdown

### PR DESCRIPTION
**Pull request by local-minimum**
This should hopefully convert block

```
!>Some important spoiler
```
into
```
<span class="spoiler">Some important spoiler</span>
```

It is fully untested as I don't know how that works.

### CODE REVIEW

**Comment by local-minimum** [comment]
It should be noted that this will treat each line starting with `!>` as a separate spoiler (if I understand things correctly). But I think that should be a good enough first iteration

**Comment by povrazor** [comment]
I'm reluctant to add this untested, because the markdown parser is perhaps the most finicky piece of code in the entire codebase.

Also you should be basing this off the `blockquote`, as the feature is effectively identical (just you'll want special `<div>` to wrap it instead of `<blockquote>`).

**Comment by local-minimum** [comment]
It sounds reasonable to be scared of adding it untested.

I'll get working on it, but it will probably take a while with playing ludum games and making a game for an other jam in parallel.

This is work on (put here to get link) #809

**Comment by zwrawr** [comment]
It also worth noting that the part of the markdown parser that is a pain is the linking, i don't think there been any issues with the rest of it. 

So don't be get scared of the evil markdown parser, the stuff your messing with shouldn't break it.

**Comment by povrazor** [comment]
Well there's a reason e-mails don't work anymore. ;)

**Comment by local-minimum** [comment]
I agree with Mike, it's better to test.
With regards to email, the regex seems wrong (doesn't allow for most mails). Should I fix that here too? And do you have an issue where this is discussed?

**Comment by zwrawr** [comment]
A separate PR would be better.
#801 is about messed up email hadling

**Comment by local-minimum** [comment]
spoilersimple

Simple non-fancy support tested and works. Will not spoil formatted stuff like links and doesn't have any nice effects on hover or click because I don't really understand `less` nor found any reasonable introduction.

**Comment by zwrawr** [comment]
Less is just some syntactical sugar on top of css. If you just write css in less it'll work.

The main thing is that less like this
```
@blue: #00f;

#content{
     div{
        color: @blue;
     }
     img{
        width: 100px;
     }
}
```

is turned into css like this
```
 #content div{
    color: #00f;
 }
 #content img{
    width: 100px;
 }
```

**Comment by local-minimum** [comment]
spoilerv2

This is how far I think I can take it. I don't understand why some urls won't get spoiled while others do.
Tested on FireFox

File: src/custom/marked/marked.js [hunk 1]
```diff
@@ -77,7 +84,7 @@ block.normal = merge({}, block);
 block.gfm = merge({}, block.normal, {
   fences: /^ *(`{3,}|~{3,})[ \.]*(\S+)? *\n([\s\S]*?)\s*\1 *(?:\n+|$)/,
   paragraph: /^/,
-  heading: /^ *(#{1,6}) +([^\n]+?) *#* *(?:\n+|$)/
+  heading: /^ *(#{1,6}) +([^\n]+?) *#* *(?:\n+|$)/,
```
Comment (zwrawr) [inline src/custom/marked/marked.js]: Is that trailing comma meant to be there?

Comment (local-minimum) [inline src/custom/marked/marked.js]: I'll fix, that was where the previous spoiler implementations went.

**Comment by local-minimum** [comment]
Just so that I know, is there anything more I should do with this or just wait for LD to be over and things settle down?

**Comment by zwrawr** [comment]
Basically wait for pov,
He's probably burnt out after LD and taking a break.
You should take a look at @Noojuno Pull Request #846 . It is a massive rework of the whole markdown and embeding code. You too a definitely going to conflict, so one of you will have to integrate your changes with theirs.

**Comment by local-minimum** [comment]
While a bit frustrating, the code in #846 is much nicer organized than what is in master and would have been easier to develop and debug in. We'll see what pov and @Noojuno says and wait.

**Comment by povrazor** [comment]
Sorry, I'm caught up with some house cleaning today. I'll hopefully get some time tonight to tackle some of the outstanding PR's.

**Comment by zwrawr** [comment]
Yea, it's bad timing that the @Noojuno finished the markdown re work soo fast. I expected it to take way longer to re write marked to be es6 complaint.

**Comment by local-minimum** [comment]
Really you should take a break pov and not stress this at all, just wanted to be sure it wasn't expected to deal with the strange links that appear in the spoiler at times.

**Comment by zwrawr** [comment]
This merge conflict is going to be fun to fix. If you get stuck jump in the chat https://gitter.im/ludumdare/ludumdare.

**Pull request closed without merging**